### ISSUE TITLE
shareObject & shareObjectRaw: TypeError: Cannot read property 'data' of undefined

### ISSUE THREAD

**Issue by adam13hylo** [pnp/pnpjs#1827]
### Category
- [ ] Enhancement
- [X] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version
2.6.0

Please specify what version(s) of SharePoint you are targeting: **SharePoint Online**

### Expected / Desired Behavior / Question
Currently trying to utilise the 'shareWith' functionality to share a file with an internal user however I am struggling to get this working. 

I have tried several variations such as:
`await sp.web.getFileByServerRelativeUrl("https://tenant.sharepoint.com/sites/sitecollection/CustomLibrary/myNewDocument.docx").shareWith("i:0#.f|membership|jbloggs@tenant.onmicrosoft.com", SharingRole.Edit);`

`await sp.web.getFileByServerRelativeUrl("/sites/sitecollection/CustomLibrary/myNewDocument.docx").shareWith("i:0#.f|membership|jbloggs@tenant.onmicrosoft.com", SharingRole.Edit);`

`await sp.web.shareObject("https://tenant.sharepoint.com/sites/sitecollection/CustomLibrary/myNewDocument.docx", "i:0#.f|membership|jbloggs@tenant.onmicrosoft.com", SharingRole.Edit);`

### Observed Behavior
Currently when trying to run the above I get the following in the console:

> TypeError: Cannot read property 'data' of undefined
    at Proxy.Queryable.configureFrom (sp.es5.umd.bundle.js:2443)
    at sendShareObjectRequest (sp.es5.umd.bundle.js:13644)
    at eval (sp.es5.umd.bundle.js:13507)
    at step (sp.es5.umd.bundle.js:318)
    at Object.eval [as next] (sp.es5.umd.bundle.js:299)
    at fulfilled (sp.es5.umd.bundle.js:289)

When running the below code using the Chrome SharePoint Editor:
`await sp.web.getFileByServerRelativeUrl("/sites/sitecollection/CustomLibrary/myNewDocument.docx").shareWith("i:0#.f|membership|jbloggs@tenant.onmicrosoft.com", SharingRole.Edit);`

If I inspect the network I can see the following:

*Request URL*
> https://tenant.sharepoint.com/sites/sitecollection/_api/web/getFileByServerRelativeUrl('%2Fsites%2Fsitecollection%2FCustomLibrary%2myNewDocument.docx')/shareObject

*Request Payload*
>includeAnonymousLinkInEmail: false
peoplePickerInput: "[{\"Key\":\"i:0#.f|membership|jbloggs@tenant.onmicrosoft.com\"}]"
propagateAcl: false
roleValue: "role:1073741827"
useSimplifiedRoles: true

*Response*
> {"odata.error":{"code":"-1, Microsoft.SharePoint.Client.ResourceNotFoundException","message":{"lang":"en-US","value":"Cannot find resource for the request shareObject."}}}

### Steps to Reproduce
I am currently developing an SPFx List View Command Set and this is where I have come across this issue, however I have tried to replicate in the 'Chrome SharePoint Editor' and I also get the same issue. I am using the following imports:

>import { sp } from "@pnp/sp";
import "@pnp/sp/sharing";
import { ISharingResult, SharingRole } from "@pnp/sp/sharing";

**Comment by patrick-rodgers** [comment]
Thanks for reporting this, we will have a look.

**Comment by patrick-rodgers** [comment]
Having a look at this and it appears that the sharing API has undergone some revisions. It has and is still beta, so that wasn't unexpected, but we haven't adjusted since these changes were made. Will do our best to go through it all before the next release, but unsure if we'll get it all covered by then. Thanks for bringing up the issue!

**Comment by patrick-rodgers** [comment]
OK, ended up fixing this bug and updating the files sharing stuff so it works.

This now works for me:
```TypeScript
const w = await sp.web.shareObject("https://tentant.sharepoint.com/sites/dev/6b6baf3e-e344-41e3-aa42-c47d4ad1be24/SharingTestLib/test.txt", "i:0#.f|membership|user@tenant.com", SharingRole.Edit);
```

As well it appears that you can no longer share files directly, you have to use the underlying item and share from there, and I have made that update as well.

Fix will go out in 2.8
